### DISCUSSION TITLE
Porting to Old Android Devices

### DISCUSSION THREAD

**Discussion by womesiete** [ggml-org/whisper.cpp#771] [Q&A]
I was able to get the whisper.android example working on the virtual Pixel in Android studio, but I wanted to see what it would take to port it to an old device. The target is an Android 9.0 Samsung Galaxy s8 with arm64-v8a CPU architecture. I imagine that I will need to find alternatives to the library dependencies in addition to keeping the minSDK below 28, setting active ABI to arm64-v8a, and modifying build.gradle with...

```
android {
    ...
    defaultConfig {
        ...
        ndk {
            abiFilters 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
        }
    }
}
```

Beyond that, I wasn't sure what would be appropriate to do or how best to investigate which specific changes need to be made. I don't have any prior experience managing the number of end user variables that exist with Android.

I get the below error when building on the device presumably because of dependency issues, but I wasn't sure how to identify the specific ones and alternatives.

```
C++ build system [configure] failed while executing:
    @echo off
    "C:\\Users\\womes\\AppData\\Local\\Android\\Sdk\\ndk\\25.1.8937393\\ndk-build.cmd" ^
      "NDK_PROJECT_PATH=null" ^
      "APP_BUILD_SCRIPT=C:\\Users\\womes\\Desktop\\Language\\Android\\whisper.cpp\\examples\\whisper.android\\app\\src\\main\\jni\\whisper\\Android.mk" ^
      "NDK_APPLICATION_MK=C:\\Users\\womes\\Desktop\\Language\\Android\\whisper.cpp\\examples\\whisper.android\\app\\src\\main\\jni\\whisper\\Application.mk" ^
      "APP_ABI=arm64-v8a" ^
      "NDK_ALL_ABIS=arm64-v8a" ^
      "NDK_DEBUG=0" ^
      "APP_PLATFORM=android-26" ^
      "NDK_OUT=C:\\Users\\womes\\Desktop\\Language\\Android\\whisper.cpp\\examples\\whisper.android\\app\\build\\intermediates\\cxx\\Release\\6e2bg5h4/obj" ^
      "NDK_LIBS_OUT=C:\\Users\\womes\\Desktop\\Language\\Android\\whisper.cpp\\examples\\whisper.android\\app\\build\\intermediates\\cxx\\Release\\6e2bg5h4/lib" ^
      "APP_SHORT_COMMANDS=false" ^
      "LOCAL_SHORT_COMMANDS=false" ^
      -B ^
      -n
  from C:\Users\womes\Desktop\Language\Android\whisper.cpp\examples\whisper.android\app
"C:\Users\womes\AppData\Local\Android\Sdk\ndk\25.1.8937393\build\..\prebuilt\windows-x86_64\bin\make.exe": Interrupt/Exception caught (code = 0xc0000005, addr = 0x00007ff737b05f9e)
```

Any help or advice would be appreciated.

**Comment by rjpikus** [comment]
You can buy Google Pixel 4a's on ebay for $75 I would do that instead of trying this. I went from an s8 to s9 to the pixel. I still did not get it running yet, in the ggml file there's a type conversion error. FWIW: arm64-v8a is the right setting for the s8 and s9 I used.
